### PR TITLE
feat: planet-dumper implementation including helm chart

### PR DESCRIPTION
**Pull request by melancholiai**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✔                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✖                                                                       |

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,13 @@
+POSTGRES_HOST=localhost
+POSTGRES_DB=openstreetmap
+POSTGRES_USER=postgres
+POSTGRES_PASSWORD=1234
+POSTGRES_PORT=5432
+DUMP_NAME_PREFIX=some_prefix
+ENABLE_OBJECT_STORAGE=true
```
Comment (CptSchnitz) [inline .env.example]:
```suggestion
UPLOAD_TO_OBJECT_STORAGE=true
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,70 @@
+FROM ubuntu:20.04 AS builder
+ARG PLANET_DUMP_NG_VERSION=1.2.0
```
Comment (CptSchnitz) [inline Dockerfile]:
```suggestion
ARG PLANET_DUMP_NG_VERSION=v1.2.0
````

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    libboost-filesystem-dev \
+    libboost-thread-dev \
+    libboost-iostreams-dev \
+    libosmpbf-dev \
+    osmpbf-bin \
+    libprotobuf-dev\
+    pkg-config \
+    curl
+
+RUN curl -L -o planet-dump-ng.tgz "https://github.com/zerebubuth/planet-dump-ng/archive/v$PLANET_DUMP_NG_VERSION.tar.gz" \
```
Comment (CptSchnitz) [inline Dockerfile]:
```suggestion
RUN curl -L -o planet-dump-ng.tgz "https://github.com/zerebubuth/planet-dump-ng/archive/$PLANET_DUMP_NG_VERSION.tar.gz" \
```

File: helm/myvalues.example.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
+global:
```
Comment (CptSchnitz) [inline helm/myvalues.example.yaml:1]: remove global

File: helm/myvalues.example.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  environment: development
+  cloudProvider:
+    name: azure
+    dockerRegistryUrl: 'some_url'
+
+enabled: true
+enableInternalObjectStorage: true
+schedule: '*/1 * * * *'
```
Comment (CptSchnitz) [inline helm/myvalues.example.yaml]: change schedule

File: helm/myvalues.example.yaml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    port: 5432
+  objectStrorage:
+    enabled: true
+    value:
+      host: hostName
+      port: 9000
+      protocol: http
+      accessKey: accessKey
+      secretKey: secretKey
+      bucketName: dumps
```
Comment (CptSchnitz) [inline helm/myvalues.example.yaml]: extract the variables out of db

File: helm/templates/cronjob.yaml [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+            - name: POSTGRES_USER
+              valueFrom:
+                secretKeyRef:
+                  name: {{ $releaseName }}-{{ $chartName }}-secret
+                  key: username
+            - name: POSTGRES_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: {{ $releaseName }}-{{ $chartName }}-secret
+                  key: password
```
Comment (CptSchnitz) [inline helm/templates/cronjob.yaml]: instead of writing each env one by one, use an entire configmap

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

File: helm/templates/cronjob.yaml (hunk 7 above)
Comment (melancholiai) [reply helm/templates/cronjob.yaml]: done.

File: helm/myvalues.example.yaml (hunk 6 above)
Comment (melancholiai) [reply helm/myvalues.example.yaml]: done.

File: helm/myvalues.example.yaml (hunk 5 above)
Comment (melancholiai) [reply helm/myvalues.example.yaml]: done.

File: helm/myvalues.example.yaml (hunk 4 above)
Comment (melancholiai) [reply helm/myvalues.example.yaml:1]: done.

File: Dockerfile (hunk 3 above)
Comment (melancholiai) [reply Dockerfile]: done.

File: Dockerfile (hunk 2 above)
Comment (melancholiai) [reply Dockerfile]: done.

File: .env.example (hunk 1 above)
Comment (melancholiai) [reply .env.example]: done.

**Comment by syncush** [comment]
@melancholiai please mark done on @CptSchnitz comments

**Review by CptSchnitz — APPROVED** [review]

**Pull request merged**